rgw/notifications: add support for sharded persistent bucket notifications (#64018)
ceph/ceph: doc/radosgw/notifications.rst

.. tip:: To minimize the latency added by asynchronous notification, we 
   recommended placing the "log" pool on fast media.

The following are the global configurations for asynchronous notifications:

.. confval:: rgw_bucket_persistent_notif_num_shards

.. note:: During upgrade scenarios, if you create a topic from a new version of radosgw, 
   there will be no guarantee on ordering of notifications on any bucket mapped to that topic.
   
.. tip:: It is also recommended that you avoid modifying or deleting topics created during 

>    upgrade scenarios, as this might result in unused RADOS objects that will not be cleaned up.

Reviewer: ```suggestion
   upgrades, as this might result in orphan RADOS objects that will not be deleted when the topic is deleted.
```

.. tip:: To minimize the latency added by asynchronous notification, we 
   recommended placing the "log" pool on fast media.

> The following are the global configurations for asynchronous notifications:

Reviewer 2: There's only one config value, so I think a list is overkill, unless there will be more in the future.  Suggest:

Persistent bucket notifications are managed by the following central configuration options:

.. confval:: rgw_bucket_persistent_notif_num_shards

> .. note:: For topics created during upgrade, per-key reordering of notifications may

Reviewer 2: A topic might be created by a Ceph upgrade?  Or is this a timing thing?  Suggest:

.. note:: When a topic is created during a Ceph upgrade, per-key ..
